pkgs/development/interpreters/babashka/default.nix

```
@@ -6,94 +6,109 @@
-  ];
-
-  doInstallCheck = true;
-
-  installCheckPhase = ''
-    $out/bin/bb --version | grep '${version}'
-    $out/bin/bb '(+ 1 2)' | grep '3'
-    $out/bin/bb '(vec (dedupe *input*))' <<< '[1 1 1 1 2]' | grep '[1 2]'
-  '';
-
-  # As of v1.2.174, this will remove references to ${graalvmDrv}/conf/chronology,
-  # not sure the implications of this but this file is not available in
-  # graalvm-ce anyway.
-  postInstall = ''
-    remove-references-to -t ${graalvmDrv} $out/bin/${executable}
-  '';
-
-  passthru.updateScript = writeScript "update-babashka" ''
-    #!/usr/bin/env nix-shell
-    #!nix-shell -i bash -p curl common-updater-scripts jq
-
-    set -euo pipefail
-
-    readonly latest_version="$(curl \
-      ''${GITHUB_TOKEN:+-u ":$GITHUB_TOKEN"} \
-      -s "https://api.github.com/repos/babashka/babashka/releases/latest" \
-      | jq -r '.tag_name')"
-
-    # v0.6.2 -> 0.6.2
-    update-source-version babashka "''${latest_version/v/}"
-  '';
-
-  meta = with lib; {
-    description = "A Clojure babushka for the grey areas of Bash";
-    longDescription = ''
-      The main idea behind babashka is to leverage Clojure in places where you
-      would be using bash otherwise.
-
-      As one user described it:
-
-          I’m quite at home in Bash most of the time, but there’s a substantial
-          grey area of things that are too complicated to be simple in bash, but
-          too simple to be worth writing a clj/s script for. Babashka really
-          seems to hit the sweet spot for those cases.
-
-      Goals:
-
-      - Low latency Clojure scripting alternative to JVM Clojure.
-      - Easy installation: grab the self-contained binary and run. No JVM needed.
-      - Familiarity and portability:
-        - Scripts should be compatible with JVM Clojure as much as possible
-        - Scripts should be platform-independent as much as possible. Babashka
-          offers support for linux, macOS and Windows.
-      - Allow interop with commonly used classes like java.io.File and System
-      - Multi-threading support (pmap, future, core.async)
-      - Batteries included (tools.cli, cheshire, ...)
-      - Library support via popular tools like the clojure CLI
-    '';
-    homepage = "https://github.com/babashka/babashka";
-    changelog = "https://github.com/babashka/babashka/blob/v${version}/CHANGELOG.md";
-    sourceProvenance = with sourceTypes; [ binaryBytecode ];
-    license = licenses.epl10;
-    maintainers = with maintainers; [
-      bandresen
-      bhougland
-      DerGuteMoritz
-      jlesquembre
-      thiagokokada
+    graalvmDrv = graalvmCEPackages.graalvm-ce;
+
+    executable = "bb";
+
+    nativeBuildInputs = [ removeReferencesTo ];
+
+    extraNativeImageBuildArgs = [
+      "-H:+ReportExceptionStackTraces"
+      "--no-fallback"
+      "--native-image-info"
+      "--enable-preview"
     ];
+
+    doInstallCheck = true;
+
+    installCheckPhase = ''
+      $out/bin/bb --version | grep '${version}'
```

```suggestion
    installCheckPhase = ''
      export CLJ_CONFIG="$TMP/.clojure"
      $out/bin/bb --version | grep '${version}'
```

That should fix the darwin issue

---

Hmm why would that only happen on Darwin? Feeling a bit reluctant to apply the fix without understanding that first... any clue?

---

(unfortunately, I don't have access to a macOS to poke at it myself ...)

---

yep, sorry for not explaining the reason.  

In the test phase we invoke `bb clojure --version`, which mimics Clojure CLI.  Clojure determines the path to the Clojure user configuration (e.g.: `~/.clojure`), and checks for a `deps.edn` file on that location. If there isn't any, it copies the default `deps.edn` file there. On Linux, in the Nix build sandbox, the path is `/build/.clojure/deps.edn`, which is writable. But on MacOS is `/var/empty/.clojure/deps.edn`, and the process fails because it can't write there. 

We can set `CLJ_CONFIG` to use a different path, so the process can copy the `deps.edn` file.

I wrote a gist with a deeper explanation:
https://gist.github.com/jlesquembre/ad7cb8357fe13312eb23a437b87bcd2b

---

Wow, thanks a lot for that deep dive! :bow: I took the liberty to file [an issue for `deps.clj` linking to your write up](https://github.com/borkdude/deps.clj/issues/114) and added your workaround with a comment pointing to it to the code.

While in there, I also hardened the version test by quoting the expected version and passing the `w` and `F` flags to `grep` so that we match the version exactly. Without that,  "1.11.1.1413" would also have satisfied versions which happen to be substrings of that version, e.g. "1.11.1.14". 

I think with that we're finally good to merge (provided the tests pass this time ...)

---

Addendum: The version check is also hardened in another way in that the dots aren't interpreted as "any character" anymore which would also have caused ambiguity (that's what the `F` flag is for).